src/Database/Adapter.php

```diff
@@ -242,6 +241,47 @@ abstract public function deleteAttribute(string $collection, string $id): bool;
      */
     abstract public function renameAttribute(string $collection, string $old, string $new): bool;
 
+    /**
+     * @param string $collection
+     * @param string $relatedCollection
+     * @param string $type
+     * @param bool $twoWay
+     * @param string $id
+     * @param string $twoWayKey
+     * @param string $onDelete
+     * @return bool
+     */
+    abstract public function createRelationship(string $collection, string $relatedCollection, string $type, bool $twoWay = false, string $id = '', string $twoWayKey = '', string $onDelete = Database::RELATION_MUTATE_RESTRICT): bool;
```

Comment:
```suggestion
    abstract public function createRelationship(string $collection, string $relatedCollection, string $type, bool $twoWay = false, string $key = '', string $twoWayKey = '', string $onDelete = Database::RELATION_MUTATE_RESTRICT): bool;
```

Better to call this a key since we use key in the other method signatures as well

Reply:
I think it should be $id here and the update method as well, will update the inverse

Reply:
I see 👍